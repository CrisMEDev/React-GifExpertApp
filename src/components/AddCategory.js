import React, { useState } from 'react'
import PropTypes from 'prop-types';

const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('Hola mundo');

    const handleInputChange = (event) => {
        setInputValue( event.target.value );
    };

    const handleSubmit = (event) => {
        event.preventDefault();     // Evitar el refresh del navegador con form

        if ( inputValue.trim().length > 2 ){
            // Se cambia el valor en la lista con el nuevo estado del input value
            setCategories(cats => [...cats, inputValue]);

            setInputValue('');
        }

    };

    return (
        // EL onUbmit es atrapado en el event del handleSubmit
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;
