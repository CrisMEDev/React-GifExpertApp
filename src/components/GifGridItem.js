import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ id, title, url }) => {

    if ( title.trim() === '' ) title = 'No title';
    
    return (
        <div className="card animate__animated animate__bounce">
            <img src={ url } alt={title} />
            <p>{ title }</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

GifGridItem.defaultProps = {
    title: 'No title',
}
