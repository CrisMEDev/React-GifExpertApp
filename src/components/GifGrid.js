import React, { useState, useEffect } from 'react'

import { GifGridItem } from './GifGridItem';
import { getGifs } from '../helpers/getGifs';

const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);
    
    useEffect( () => {

        getGifs( category ).then( gifs => setImages( gifs ) );

    }, [category]);


    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                {
                images.map( (gif) => <GifGridItem key={ gif.id } { ...gif } /> )
                }
            </div>
        </>
    )
}

export default GifGrid
