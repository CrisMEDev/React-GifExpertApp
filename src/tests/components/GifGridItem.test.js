import React from 'react';
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en el componente GifGridItem', () => {
    
    const data = {
        id: 123,
        title: 'Un titulo comun',
        url: 'https://myawesomeurl.com'
    };
    const wrapper = shallow( <GifGridItem {...data} /> );
    
    test('Debe de mostrar el componente <GifGridItem /> correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
        
    });

    test('Debe de tener un parrafo con el title', () => {
        
        const titulo = wrapper.find('p');

        expect( titulo.text().trim() ).toBe( data.title );

    });

    test('Debe tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');
        
        // console.log( img.html() );  // Para ver la etiqueta obtenida
        // console.log( img.props() );  // Para obtener las propiedades de la etiqueta como un objeto
        // console.log( img.prop('src) );  // Para obtener la propiedad de la etiqueta como un string

        expect( img.prop('src') ).toBe( data.url );
        
    });

    test('Debe de tener la clase animate__bounce', () => {

        const div = wrapper.find('div');

        // expect( div.prop('className').includes('animate__bounce') ).toBe(true);
        expect( div.hasClass('animate__bounce') ).toBe(true);
        
    });
    
    
    
});
