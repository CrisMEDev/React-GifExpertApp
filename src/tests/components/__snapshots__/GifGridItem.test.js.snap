// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGridItem Debe de mostrar el componente <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Un titulo comun"
    src="https://myawesomeurl.com"
  />
  <p>
    Un titulo comun
  </p>
</div>
`;
